generator/sponsored_dataset.hpp

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include "indexer/index.hpp"
+
+#include "search/reverse_geocoder.hpp"
+
+#include "base/newtype.hpp"
+
+#include "std/function.hpp"
+#include "std/limits.hpp"
+#include "std/map.hpp"
+#include "std/string.hpp"
+
+#include "boost/geometry.hpp"
+#include "boost/geometry/geometries/point.hpp"
+#include "boost/geometry/geometries/box.hpp"
+#include "boost/geometry/index/rtree.hpp"
+
+class FeatureBuilder1;
+
+namespace generator
+{
+class SponsoredDataset
+{
+public:
+  NEWTYPE(uint32_t, ObjectId);
+
+  static double constexpr kDistanceLimitInMeters = 150;
+  static size_t constexpr kMaxSelectedElements = 3;
+  static ObjectId const kInvalidObjectId;
+
+  struct Object
+  {
+    enum class Fields
+    {
+      Id = 0,
+      Latitude = 1,
+      Longtitude = 2,
+      Name = 3,
+      Address = 4,
+      Stars = 5,
+      PriceCategory = 6,
+      RatingBooking = 7,
+      RatingUsers = 8,
+      DescUrl = 9,
+      Type = 10,
+      Translations = 11,
+
+      Counter
+    };
+
+    ObjectId id{kInvalidObjectId};
+    double lat = 0.0;
+    double lon = 0.0;
+    string name;
+    string address;
+    string street;
+    string houseNumber;
+    uint32_t stars = 0;
+    uint32_t priceCategory = 0;
+    double ratingBooking = 0.0;
+    double ratingUser = 0.0;
+    string descUrl;
+    uint32_t type = 0;
+    string translations;
+
+    static constexpr size_t Index(Fields field) { return static_cast<size_t>(field); }
+    static constexpr size_t FieldsCount() { return static_cast<size_t>(Fields::Counter); }
+    explicit Object(string const & src);
```

Comment:
Please, insert a blank line between ctor and FieldsCount static method. Also, please move all statics below ctor - ctor should go first.